### PR TITLE
Add Missing Methods

### PR DESCRIPTION
**Pull request by hryvinskyi**
### Description
Add Missing Methods

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @hryvinskyi. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me {$VERSION} instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Comment by ishakhsuvarov** [comment]
Hi @hryvinskyi 
Why do you think that these methods are missing? Could you please provide more background for this fix?

Thank you

**Comment by hryvinskyi** [comment]
Hi @ishakhsuvarov
Many of them are used in the code, but they are not explicitly implemented in the class. This does not allow you to write a plugin for them. And also, in my opinion, all methods should be explicit.

Sorry for my bad english :)

File: app/code/Magento/Quote/Model/Quote/Item/AbstractItem.php [hunk 1]
```diff
@@ -599,6 +603,26 @@ public function setConvertedPrice($value)
...
     }
 
+    /**
+     * @param bool $status
+     *
+     * @return AbstractItem
+     */
+    public function setHasChildren(bool $status): AbstractItem
+    {
+        $this->setData('has-children', $status);
```
Comment (orlangur) [inline app/code/Magento/Quote/Model/Quote/Item/AbstractItem.php:613]: Such change is backward-incompatible, by the way...

I see `$this->setHasChildren(true);` in code, there is no need in explicit methods implementation, just add them as `@method` annotation.

**Review by orlangur — CHANGES_REQUESTED** [review]

**Comment by sidolov** [comment]
Hi @hryvinskyi , I'm closing this event as not relevant according to comment from @orlangur. 
Thank you for the contribution!

**Pull request closed without merging**